airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -85,18 +89,40 @@ def __get__(self, instance, owner):
         return self._metadata
 
 
+class GoogleAnalyticsDataApiBackoffStrategy(BackoffStrategy):
+    def backoff_time(
+        self, response_or_exception: Optional[Union[requests.Response, requests.RequestException]], **kwargs: Any
+    ) -> Optional[float]:
+        if isinstance(response_or_exception, requests.Response):
+            # handle the error with prepared GoogleAnalyticsQuotaHandler backoff value
+            if response_or_exception.status_code == requests.codes.too_many_requests:
+                return GoogleAnalyticsQuotaHandler.backoff_time
+        return None
+
+
+class GoogleAnalyticsDatApiErrorHandler(HttpStatusErrorHandler):
+    @GoogleAnalyticsQuotaHandler.handle_quota()
+    def interpret_response(self, response_or_exception: Optional[Union[requests.Response, Exception]] = None) -> ErrorResolution:
+        if response_or_exception.status_code == requests.codes.too_many_requests:
+            return ErrorResolution(
+                response_action=GoogleAnalyticsQuotaHandler.response_action,
+                failure_type=FailureType.transient_error,
+                error_message=GoogleAnalyticsQuotaHandler.error_message,
+            )
```

Comment:
This is related to green weeks because this connector currently raises a lot of system errors for 429 rate limited which should actually be recategorized as transient errors.